pkg/datapath/linux/ipsec/ipsec_linux.go

```diff
@@ -925,6 +913,25 @@ func LoadIPSecKeys(r io.Reader) (int, uint8, error) {
 	return keyLen, spi, nil
 }
 
+func parseSPI(spiStr string) (uint8, int, error) {
+	offsetBase := 0
+	spi, err := strconv.Atoi(spiStr)
+	if err != nil {
+		// If no version info is provided assume using key format without
+		// versioning and assign SPI.
+		log.Warning("IPsec secrets without an SPI as the first argument are deprecated and will be unsupported in v1.13.")
+		spi = 1
```

Comment:
Since we are already refactoring we could add an early return here:
```suggestion
		return 1, -1, nil
```